### PR TITLE
Replace material-ui-popup-state with custom menu

### PR DESCRIPTION
**Pull request by tjennison-work**
It requires clunky code and interacts poorly with other theming.

### CODE REVIEW

File: ui/src/menu.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  useCallback,
+  useState,
+} from "react";
+
+interface MenuProps
+  extends Omit<BaseMenuProps, "anchorEl" | "open" | "onClick" | "onClose"> {
+  items: ReactNode;
+}
+
+export function useMenu(
```
Comment (calbach) [inline ui/src/menu.tsx:19]: Interesting, are you basing this on an existing pattern or style guide from somewhere? I've not seen this "hook returning a component" pattern before, but from searching around I do see some mentions of it online.

**Review by calbach — APPROVED** [review]

File: ui/src/overview.tsx [hunk 2]
```diff
@@ -140,36 +130,36 @@ function ParticipantsGroup(props: { group: Group }) {
...
-    variant: "popover",
-    popupId: "criteria",
-  });
-
   const updater = useDatasetUpdater();
 
+  const [menu, show] = useMenu({
+    items: [
+      <MenuItem
+        key="1"
```
Comment (cbookg) [inline ui/src/overview.tsx:138]: do you need these keys?

File: ui/src/overview.tsx [hunk 3]
```diff
@@ -98,25 +86,27 @@ function AddCriteriaButton(props: { group: string | GroupKind }) {
     },
   ];
 
+  const [menu, show] = useMenu({
+    items: items.map((item) => {
+      return (
```
Comment (cbookg) [inline ui/src/overview.tsx:91]: can be reduced to:  (item) => (
  <MenuItem  ... >
)

File: ui/src/menu.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  props: MenuProps
+): [ReactElement, (e: MouseEvent<HTMLElement | undefined>) => void] {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | undefined>();
+  const show = useCallback((e) => setAnchorEl(e.currentTarget), []);
+
+  const menu = useCallback(
+    (props: MenuProps): ReactElement => {
+      return (
+        <>
```
Comment (cbookg) [inline ui/src/menu.tsx:28]: drop this and use BaseMenu as top level?

File: ui/src/menu.tsx [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+}
+
+export function useMenu(
+  props: MenuProps
+): [ReactElement, (e: MouseEvent<HTMLElement | undefined>) => void] {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | undefined>();
+  const show = useCallback((e) => setAnchorEl(e.currentTarget), []);
+
+  const menu = useCallback(
+    (props: MenuProps): ReactElement => {
```
Comment (cbookg) [inline ui/src/menu.tsx:26]: just => ( <element/> )

File: ui/src/menu.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+          >
+            {props.items}
+          </BaseMenu>
+        </>
+      );
+    },
+    [props]
+  );
+
+  return [createElement(menu, props, null), show];
```
Comment (cbookg) [inline ui/src/menu.tsx:44]: im trying to understand this, but not quite following based on the docs, which have the first param being a type (but you're padding a callback)?

mainly curious when the callback would be invoked, and how its not run each time.

**Pull request merged**